### ISSUE TITLE
Password login not possible after the password has set/updated via bcozy.

### ISSUE THREAD

**Issue by DivineThreepwood** [openbase/bco.authentication#76]
When setting the admin password back to "admin" at the bcozy default password change prompt the password seems not to be stored correctly. After this change, no admin login is any longer possible at all. BCO prints the following log message after each try:
```
15:46:18.034 [pool-1-thread-124] ERROR o.o.b.a.c.AuthenticatorController ═╦═⚀ Fatal implementation error in or by using class org.openbase.bco.authentication.lib.EncryptionHelper: Unable to encrypt object[[B@46b8b084]
15:46:18.034 [pool-1-thread-124] ERROR o.o.b.a.c.AuthenticatorController  ╚════⚀ Invalid AES key length: 48 bytes
15:46:18.034 [pool-1-thread-124] ERROR o.o.b.a.c.AuthenticatorController =====================================
15:46:18.034 [pool-1-thread-124] ERROR o.o.b.a.c.AuthenticatorController Fatal implementation error in or by using class org.openbase.bco.authentication.lib.EncryptionHelper: Unable to encrypt object[[B@46b8b084]
org.openbase.jul.exception.FatalImplementationErrorException: Fatal implementation error in or by using class org.openbase.bco.authentication.lib.EncryptionHelper: Unable to encrypt object[[B@46b8b084]
	at org.openbase.bco.authentication.lib.EncryptionHelper.encrypt(EncryptionHelper.java:194)
	at org.openbase.bco.authentication.lib.EncryptionHelper.encrypt(EncryptionHelper.java:157)
	at org.openbase.bco.authentication.lib.AuthenticationServerHandler.handleKDCRequest(AuthenticationServerHandler.java:93)
	at org.openbase.bco.authentication.core.AuthenticatorController.lambda$requestTicketGrantingTicket$0(AuthenticatorController.java:247)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.security.InvalidKeyException: Invalid AES key length: 48 bytes
	at java.base/com.sun.crypto.provider.AESCrypt.init(AESCrypt.java:91)
	at java.base/com.sun.crypto.provider.ElectronicCodeBook.init(ElectronicCodeBook.java:94)
	at java.base/com.sun.crypto.provider.CipherCore.init(CipherCore.java:591)
	at java.base/com.sun.crypto.provider.CipherCore.init(CipherCore.java:467)
	at java.base/com.sun.crypto.provider.AESCipher.engineInit(AESCipher.java:313)
	at java.base/javax.crypto.Cipher.implInit(Cipher.java:839)
	at java.base/javax.crypto.Cipher.chooseProvider(Cipher.java:901)
	at java.base/javax.crypto.Cipher.init(Cipher.java:1286)
	at java.base/javax.crypto.Cipher.init(Cipher.java:1223)
	at org.openbase.bco.authentication.lib.EncryptionHelper.encrypt(EncryptionHelper.java:183)
	... 7 common frames omitted
15:46:18.040 [pool-1-thread-124] ERROR o.o.b.a.c.AuthenticatorController =====================================
```
and
```
15:32:33.980 [pool-1-thread-5] ERROR o.o.b.a.c.AuthenticatorController =====================================
15:32:33.982 [pool-1-thread-180] ERROR o.o.b.a.c.AuthenticatorController ═╦═⚀ Could not invoke Method[Class requestTicketGrantingTicket(UserClientPair)] of org.openbase.bco.authentication.core.AuthenticatorController@777413d6!
15:32:33.983 [pool-1-thread-180] ERROR o.o.b.a.c.AuthenticatorController  ╚════⚀ Internal server error. Please try again.
15:32:33.983 [pool-1-thread-180] ERROR o.o.b.a.c.AuthenticatorController =====================================
15:32:33.983 [pool-1-thread-180] ERROR o.o.b.a.c.AuthenticatorController Could not invoke Method[Class requestTicketGrantingTicket(UserClientPair)] of org.openbase.bco.authentication.core.AuthenticatorController@777413d6!
org.openbase.jul.exception.CouldNotPerformException: Could not invoke Method[Class requestTicketGrantingTicket(UserClientPair)] of org.openbase.bco.authentication.core.AuthenticatorController@777413d6!
	at org.openbase.jul.communication.controller.RPCHelper.lambda$registerMethod$1(RPCHelper.java:167)
	at rsb.patterns.LocalMethod.internalNotify(LocalMethod.java:121)
	at org.openbase.jul.extension.rsb.com.strategy.ThreadPoolUnorderedEventReceivingStrategy$DispatchTask.call(ThreadPoolUnorderedEventReceivingStrategy.java:198)
	at org.openbase.jul.extension.rsb.com.strategy.ThreadPoolUnorderedEventReceivingStrategy$DispatchTask.call(ThreadPoolUnorderedEventReceivingStrategy.java:169)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.util.concurrent.ExecutionException: org.openbase.jul.exception.CouldNotPerformException: Internal server error. Please try again.
	at java.base/java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:205)
	at org.openbase.jul.communication.controller.RPCHelper.lambda$registerMethod$1(RPCHelper.java:143)
	... 7 common frames omitted
Caused by: org.openbase.jul.exception.CouldNotPerformException: Internal server error. Please try again.
	at org.openbase.bco.authentication.core.AuthenticatorController.lambda$requestTicketGrantingTicket$0(AuthenticatorController.java:254)
	... 4 common frames omitted
15:32:34.042 [pool-1-thread-180] ERROR o.o.b.a.c.AuthenticatorController =====================================
```

**Comment by DivineThreepwood** [comment]
Same effect when changing the password via ``bco-console``.

**Comment by pLeminoq** [comment]
Fixed in commit dd49d9a077d2cd8cdbd1b3acc4873be63f8a9d36. There was an error in the changeCredentials method of the session manager which caused the authenticator to save new credentials and check old credentials encrypted with the current session key. You would not even have been able to change the password if you tried with a normal user. But for admins the check if the old credentials are valid is skipped.
In the commit which fixes the issue I also added a unit test for the changeCredentials method of the session manager.
